frontend/catalyst/jax_extras/tracing.py

```diff
@@ -248,6 +312,97 @@ def __init__(self, boxid: int, e: JaxprEqn):
     return [b.e for b in stable_toposort(boxes)]  # [4]
 
 
+@transformation_with_aux
+def expanded_fun(static_args, *args_expanded):
+    """Function transformation making the function to accept its arguments in the expanded format
+    (with the dimension variables added)"""
+    (in_type, expansion_strategy) = static_args
+    args_collapsed = [a for a, (_, k) in zip(args_expanded, in_type) if k]
+    res_flat = yield args_collapsed, {}
```

Comment:
What is this dictionary that we also return empty?

Reply:
These transformations work on a quite sophisticated engine defined in Jax libraries. See for [this comment](https://github.com/google/jax/blob/88a60b808c1f91260cc9e75b9aa2508aae5bc9f9/jax/_src/linear_util.py#L49) for the API description, I think it is quite stable.
The idea, as I understand it, is to define wrappers being able to pre-process arguments and post-process results without access to the underlying function reference.
The empty dictionary stands for kwargs, which we already encode into positional args [here](https://github.com/PennyLaneAI/catalyst/blob/24b836ffc1b7f38fe9653d580a1d6abfec2cf390/frontend/catalyst/jax_extras/tracing.py#L382)

Reply:
Thanks for adding pointing to this comment in the JAX source code. Can you point to that comment here somewhere in this function as well? As in, just write a comment that links to that comment.

Reply:
Ok, should be done.